webapp/src/ts/services/form.service.ts

```diff
@@ -326,7 +331,39 @@ export class FormService {
     }, null);
   }
 
-  async saveContact(form, docId, type, xmlVersion) {
+  async checkForDuplicates(doc, duplicateCheck, acknowledged) {
+    const parentId = doc ? doc.parent?._id : undefined;
+    const contactType = doc ? doc.contact_type ?? doc.type : undefined;
+    const siblings = await requestSiblings(this.dbService, parentId, contactType);
```

Comment:
issue: we should definitely move this down after the if-check below since we do not want to query the contact siblings unless we are really going to be doing the dupe check.